### PR TITLE
chore: fix docstrings in preparation for `doctest`

### PR DESCRIPTION
**Pull request by andrijapau**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane/noise/add_noise.py [hunk 1]
```diff
@@ -104,16 +104,14 @@ def circuit(w, x, y, z):
...
-        1: ──RY(0.40)──────────────────────────────────────────────────────────────────╰X──RX(0.60)
-
-        ────────────────────────────────────────────────────────────────────PhaseFlip(0.2)─┤ ╭<Z@Z>
-        ───PhaseDamping(0.40)──ThermalRelaxationError(0.30,2.00,0.20,0.60)──PhaseFlip(0.2)─┤ ╰<Z@Z>
+    >>> circuit(0.9, 0.4, 0.5, 0.6)
+    array(0.544053)
+    >>> print(qml.draw(circuit)(0.9, 0.4, 0.5, 0.6))
+    0: ──RX(0.90)──PhaseDamping(0.40)──ThermalRelaxationError(0.45,2.00,0.20,0.60)─╭●──RY(0.50)
+    1: ──RY(0.40)──────────────────────────────────────────────────────────────────╰X──RX(0.60)
+
```
Comment (andrijapau) [inline pennylane/noise/add_noise.py]:
```suggestion
    <BLANKLINE>
```

File: pennylane/qcut/cutcircuit_mc.py [hunk 2]
```diff
@@ -607,39 +607,37 @@ def expand_fragment_tapes_mc(
...
+    >>> print(settings)
+    [[1 6 2]]
+    >>> for i, (c1, c2) in enumerate(zip(configs[0], configs[1])):
+    ...     print(f"config {i}:")
+    ...     print(c1.draw())
+    ...     print("")
+    ...     print(c2.draw())
+    ...     print("")
+    ...
+
```
Comment (andrijapau) [inline pennylane/qcut/cutcircuit_mc.py]:
```suggestion
    <BLANKLINE>
```

File: pennylane/templates/swapnetworks/ccl2.py [hunk 3]
```diff
@@ -45,46 +45,42 @@ class TwoLocalSwapNetwork(Operation):
...
+        >>> print(weights)
+        tensor([0.20308242, 0.91906199, 0.67988804, 0.81290256, 0.08708985,
+                0.81860084, 0.34448344, 0.05655892, 0.61781612, 0.51829044], requires_grad=True)
+        >>> acquaintances = lambda index, wires, param: (qml.CRY(param, wires=index)
+        ...                                  if np.abs(wires[0]-wires[1]) else qml.CRZ(param, wires=index))
+        >>> @qml.qnode(dev)
+        ... def swap_network_circuit():
+        ...    qml.templates.TwoLocalSwapNetwork(dev.wires, acquaintances, weights, fermionic=False)
+        ...    return qml.state()
+        >>> qml.draw(swap_network_circuit, level='device')()
```
Comment (isaacdevlugt) [inline pennylane/templates/swapnetworks/ccl2.py]:
```suggestion
        >>> print(qml.draw(swap_network_circuit, level='device')())
```

File: pennylane/workflow/qnode.py [hunk 4]
```diff
@@ -453,9 +451,9 @@ def circuit(x, y, U):
                 qml.RY(y, wires=3)
                 return qml.expval(qml.Z(0) @ qml.X(1) @ qml.Z(2) @ qml.Z(3))
 
-
             x = np.array([0.4, 2.1, -1.3])
             y = 2.71
+            np.random.seed(42)
```
Comment (isaacdevlugt) [inline pennylane/workflow/qnode.py]: pourquoi?

File: pennylane/gradients/metric_tensor.py [hunk 5]
```diff
@@ -356,23 +356,24 @@ def circuit(weights):
...
+                return qml.expval(qml.Z(0))
+
+            weights = np.array([0.1, 0.2, 0.4, 0.5], requires_grad=True)
+            mt = qml.metric_tensor(circuit, argnum=(0, 2, 3))(weights)
+
+        >>> print(mt)
+        [[ 0.          0.          0.          0.        ]
+            [ 0.          0.25       -0.02495835 -0.02495835]
+            [ 0.         -0.02495835  0.01226071  0.01226071]
+            [ 0.         -0.02495835  0.01226071  0.01226071]]
```
Comment (isaacdevlugt) [inline pennylane/gradients/metric_tensor.py]: Indentation is off here I believe

File: pennylane/noise/add_noise.py [hunk 6]
```diff
@@ -104,16 +104,14 @@ def circuit(w, x, y, z):
...
-        1: ──RY(0.40)──────────────────────────────────────────────────────────────────╰X──RX(0.60)
-
-        ────────────────────────────────────────────────────────────────────PhaseFlip(0.2)─┤ ╭<Z@Z>
-        ───PhaseDamping(0.40)──ThermalRelaxationError(0.30,2.00,0.20,0.60)──PhaseFlip(0.2)─┤ ╰<Z@Z>
+    >>> circuit(0.9, 0.4, 0.5, 0.6)
+    array(0.544053)
+    >>> print(qml.draw(circuit)(0.9, 0.4, 0.5, 0.6))
+    0: ──RX(0.90)──PhaseDamping(0.40)──ThermalRelaxationError(0.45,2.00,0.20,0.60)─╭●──RY(0.50)
+    1: ──RY(0.40)──────────────────────────────────────────────────────────────────╰X──RX(0.60)
+    <BLANKLINE>
```
Comment (isaacdevlugt) [inline pennylane/noise/add_noise.py:112]: Is this `blankline` thing supposed to be there? 🤔

File: pennylane/ops/qubit/arithmetic_ops.py [hunk 7]
```diff
@@ -69,11 +69,13 @@ class QubitCarry(Operation):
...
 
-    .. code-block::
+    .. code-block:: python
+
+        import itertools
 
         input_bitstring = (0, 1, 1, 0)
 
-        @qml.qnode(dev)
+        @qml.qnode(qml.device("default.qubit", wires=4))
```
Comment (isaacdevlugt) [inline pennylane/ops/qubit/arithmetic_ops.py]: 🫢

File: pennylane/qcut/cutcircuit_mc.py [hunk 8]
```diff
@@ -607,39 +607,37 @@ def expand_fragment_tapes_mc(
...
+    >>> print(settings)
+    [[1 6 2]]
+    >>> for i, (c1, c2) in enumerate(zip(configs[0], configs[1])):
+    ...     print(f"config {i}:")
+    ...     print(c1.draw())
+    ...     print("")
+    ...     print(c2.draw())
+    ...     print("")
+    ...
+    <BLANKLINE>
```
Comment (isaacdevlugt) [inline pennylane/qcut/cutcircuit_mc.py:620]: blankline?

File: pennylane/noise/add_noise.py (hunk 6 above)
Comment (andrijapau) [reply pennylane/noise/add_noise.py:112]: If we want separation then yes

File: pennylane/qcut/cutcircuit_mc.py (hunk 8 above)
Comment (andrijapau) [reply pennylane/qcut/cutcircuit_mc.py:620]: If we want separation then yes

File: pennylane/workflow/qnode.py (hunk 4 above)
Comment (andrijapau) [reply pennylane/workflow/qnode.py]: The unitary group bit is random so if we want the docs to always be true we need to seed. Optionally we re-write it with a specific matrix from that group and dodge the problem all together. Thoughts?

File: doc/releases/changelog-dev.md [hunk 9]
```diff
@@ -1272,6 +1272,9 @@
 
 <h3>Documentation 📝</h3>
 
+* Fix docstrings to not have `>>>` and `...` inside of `.. code-block::` directives.
```
Comment (JerryChen97) [inline doc/releases/changelog-dev.md]:
```suggestion
* Remove `>>>` and `...` from `.. code-block::` directives in docstrings.
```

File: pennylane/gradients/metric_tensor.py (hunk 5 above)
Comment (andrijapau) [reply pennylane/gradients/metric_tensor.py]: Good catch, fixed here https://github.com/PennyLaneAI/pennylane/pull/8319/commits/14d4354fdf9f6f7009696fb32d85ff5bad083fd8

File: pennylane/ops/qubit/arithmetic_ops.py (hunk 7 above)
Comment (andrijapau) [reply pennylane/ops/qubit/arithmetic_ops.py]: yeah ... lots of our docstrings aren't executable :heh:

Comment (albi3ro) [reply pennylane/ops/qubit/arithmetic_ops.py]: Do we need to specify wires? or would `qml.device('default.qubit')` get the job done?

File: pennylane/ops/qubit/arithmetic_ops.py [hunk 10]
```diff
@@ -237,11 +239,13 @@ class QubitSum(Operation):
...
 
-    .. code-block::
+    .. code-block:: python
+
+        import itertools
 
         input_bitstring = (0, 1, 0)
 
-        @qml.qnode(dev)
+        @qml.qnode(qml.device("default.qubit", wires=3))
```
Comment (albi3ro) [inline pennylane/ops/qubit/arithmetic_ops.py]: same here.  Do we need the wires?

File: pennylane/workflow/qnode.py (hunk 4 above)
Comment (albi3ro) [reply pennylane/workflow/qnode.py]: Can you use a local seed instead?

File: pennylane/ops/qubit/arithmetic_ops.py (hunk 10 above)
Comment (andrijapau) [reply pennylane/ops/qubit/arithmetic_ops.py]: Nope, I can change it

File: pennylane/ops/qubit/arithmetic_ops.py (hunk 7 above)
Comment (andrijapau) [reply pennylane/ops/qubit/arithmetic_ops.py]: Nope, I can change it

File: pennylane/gradients/metric_tensor.py [hunk 11]
```diff
@@ -356,23 +356,24 @@ def circuit(weights):
...
-            ...     return qml.expval(qml.Z(0))
-
-            >>> weights = np.array([0.1, 0.2, 0.4, 0.5], requires_grad=True)
-            >>> mt = qml.metric_tensor(circuit, argnum=(0, 2, 3))(weights)
-            >>> print(mt)
-            [[ 0.          0.          0.          0.        ]
-             [ 0.          0.25       -0.02495835 -0.02495835]
-             [ 0.         -0.02495835  0.01226071  0.01226071]
-             [ 0.         -0.02495835  0.01226071  0.01226071]]
+            dev = qml.device("default.qubit", wires=3)
```
Comment (andrijapau) [inline pennylane/gradients/metric_tensor.py]:
```suggestion
            dev = qml.device("default.qubit")
```

File: pennylane/ops/qubit/arithmetic_ops.py (hunk 7 above)
Comment (andrijapau) [inline pennylane/ops/qubit/arithmetic_ops.py]:
```suggestion
        @qml.qnode(qml.device("default.qubit"))
```

File: pennylane/ops/qubit/arithmetic_ops.py (hunk 10 above)
Comment (andrijapau) [inline pennylane/ops/qubit/arithmetic_ops.py]:
```suggestion
        @qml.qnode(qml.device("default.qubit"))
```

Comment (andrijapau) [reply pennylane/ops/qubit/arithmetic_ops.py]: https://github.com/PennyLaneAI/pennylane/pull/8319/commits/4f3883a8d98ef7ae51662183a75582791edd49b9

File: pennylane/ops/qubit/arithmetic_ops.py (hunk 7 above)
Comment (andrijapau) [reply pennylane/ops/qubit/arithmetic_ops.py]: https://github.com/PennyLaneAI/pennylane/pull/8319/commits/4f3883a8d98ef7ae51662183a75582791edd49b9

File: pennylane/workflow/qnode.py (hunk 4 above)
Comment (andrijapau) [inline pennylane/workflow/qnode.py]:
```suggestion
```

Comment (andrijapau) [reply pennylane/workflow/qnode.py]: I'll remove it for now

Comment (andrijapau) [reply pennylane/workflow/qnode.py]: I decided to just revert the qnode.py changes since they are not needed to make doctest work.

File: pennylane/noise/add_noise.py [hunk 12]
```diff
@@ -104,16 +104,14 @@ def circuit(w, x, y, z):
...
-        array(0.544053)
-        >>> print(qml.draw(circuit)(0.9, 0.4, 0.5, 0.6))
-        0: ──RX(0.90)──PhaseDamping(0.40)──ThermalRelaxationError(0.45,2.00,0.20,0.60)─╭●──RY(0.50)
-        1: ──RY(0.40)──────────────────────────────────────────────────────────────────╰X──RX(0.60)
-
-        ────────────────────────────────────────────────────────────────────PhaseFlip(0.2)─┤ ╭<Z@Z>
-        ───PhaseDamping(0.40)──ThermalRelaxationError(0.30,2.00,0.20,0.60)──PhaseFlip(0.2)─┤ ╰<Z@Z>
+    >>> circuit(0.9, 0.4, 0.5, 0.6)
+    array(0.544053)
+    >>> print(qml.draw(circuit)(0.9, 0.4, 0.5, 0.6))
```
Comment (AntonNI8) [inline pennylane/noise/add_noise.py:109]: This example could be updated since `draw` has changed a bit.

Comment (andrijapau) [reply pennylane/noise/add_noise.py:109]: Hopefully we'll have a tool in-place to detect that :)

File: pennylane/templates/swapnetworks/ccl2.py [hunk 13]
```diff
@@ -45,46 +45,42 @@ class TwoLocalSwapNetwork(Operation):
...
-        3: ─╰X─╰fSWAP(3.14)─╭●─╭fSWAP(3.14)─╰X─╰fSWAP(3.14)─╭●─╭fSWAP(3.14)─╰X─╰fSWAP(3.14)─┤  State
-        4: ─────────────────╰X─╰fSWAP(3.14)─────────────────╰X─╰fSWAP(3.14)─────────────────┤  State
+    >>> import pennylane as qml
+    >>> dev = qml.device('default.qubit', wires=5)
+    >>> acquaintances = lambda index, wires, param=None: qml.CNOT(index)
+    >>> @qml.qnode(dev)
+    ... def swap_network_circuit():
+    ...    qml.templates.TwoLocalSwapNetwork(dev.wires, acquaintances, fermionic=True, shift=False)
+    ...    return qml.state()
+    >>> qml.draw(swap_network_circuit, level='device')()
```
Comment (isaacdevlugt) [inline pennylane/templates/swapnetworks/ccl2.py]:
```suggestion
    >>> print(qml.draw(swap_network_circuit, level='device')())
```
